prog/prog.go

```diff
@@ -116,6 +116,14 @@ func (a *Arg) Size() uintptr {
 				size += fld.Size()
 			}
 		}
+		align := typ.Align()
+		if size % align != 0 {
```

Comment:
I would write this as:

```go
align := typ.Align()
if typ.Varlen {
	size += align - size%align
}
if size%align != 0 {
	panic(fmt.Sprintf("struct %+v with type %+v has static size %v, which isn't aligned to %v", a, typ, size, align))
}
```

Because then we also check that `size += align - size%align` calculation is correct.